docs/adr/0008-redis-lua-scripts.md

```diff
@@ -0,0 +1,69 @@
+# Redis Lua Scripts
+
+- Status: draft
+- Deciders: dannycoates + TBD
+- Date: 2019-11-15
+
+## Context and Problem Statement
+
+We're planning on expanding our use of Redis to replace some or all of our MySQL database for OAuth. Our `fxa-shared/redis` module also requires maintenance as part of our modernization effort. The expansion will include use cases that require new indexing, join-like, and transactional operations.
```

Comment:
It would be good to include a more thorough background for what problems moving the data from MySQL to Redis will help solve and how we got to the decision to use Redis more. There's a lot of implied contextual knowledge that is skipped here, which would be great to capture. e.g. that our `fxa-shared/redis` module is currently capturing application logic that would be better handled in Redis.

Reply:
That MySQL to Redis probably deserves its own ADR. This one is specifically about switching to lua scripts for transactional ops. 
... 
_wait, did i just create more work for myself_🤦‍♂️

Reply:
Ah, okey, good point. Then maybe just background on what application logic or issues we have now that the Lua scripts will resolve.